soc/arm/common/cortex_m/arm_mpu_regions.c

```diff
@@ -30,6 +30,12 @@ static const struct arm_mpu_region mpu_regions[] = {
 			 REGION_RAM_ATTR(REGION_SRAM_SIZE)),
 #endif
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_qspi)
+	MPU_REGION_ENTRY("QUADSPI",
+			 QSPI_BASE,
+			 (STRONGLY_ORDERED_SHAREABLE | REGION_256M | NO_ACCESS_Msk)),
+#endif
+
```

Comment:
The point of having mpu_regions.c files in soc/arm/<vnd>/ is to avoid each soc putting its specifics into this common file.
So if we'd want to apply this to F7 series, we'd need to create a `soc/arm/st_stm32/stm32f7/mpu_regions.c` new file.
And this is exactly to avoid this that https://github.com/zephyrproject-rtos/zephyr/pull/43119 was created